### PR TITLE
Edit and inspect binaryrelationship in relationshipmatrix fixes #144

### LINKED ISSUE #144: RelationshipMatrix: as a user i want to inspect/edit a relationship from the intersection cell of 2 things
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the CDP4-IME
- [x] I have searched [open](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
A user can create/delete relationships using the relationship matrix. Currently the user cannot inspect or edit the selected relationship from the matrix. This shall be possible

### Steps to Reproduce
 - open relationship matrix and configure in such a way that a relationship is visible.
 - right-mouse click on a relationship -> see that it is only possible to create / delete and not inspect.

### Logs
NA
  
### System Configuration

- CDP4 version:
  - [x] CDP4-IME: 3.1.1

### PR DESCRIPTION
**Pull request by lxatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/pulls) open
- [x] I have verified that I am following the CDP4-IME [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/CDP4-IME-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
Added functionality to edit and inspect binaryrelationships in relationshipmatrix.
When multiple binaryrelationships, in the same direction, exist for the two elements, then only the first one (LINQ) will be shown in the edit / inspect Dialog window.

A unit test was failing unexpectedly, which in the end turned out to be caused at an earlier stage (mea culpa). 
This was also fixed in this pull request along with some boyscouting/refactoring concerning that particular piece of code (FilterStringService).

### CODE REVIEW

File: RelationshipMatrix/ViewModels/RelationshipMatrixViewModel.cs [hunk 1]
```diff
@@ -273,6 +293,43 @@ public bool CanEditSourceX
...
+
+        /// <summary>
+        /// Gets or sets a value indicating whether a <see cref="BinaryRelationship" /> can be inspected from sourceY to sourceX
+        /// </summary>
+        public bool CanInspectSourceXToSourceY
+        {
+            get { return this.canInspectSourceXToSourceY; }
+            private set { this.RaiseAndSetIfChanged(ref this.canInspectSourceXToSourceY, value); }
+        }
+
```
Comment (alexatrhea) [inline RelationshipMatrix/ViewModels/RelationshipMatrixViewModel.cs]: im 💯 % sure that's a extra blank line 👎

File: RelationshipMatrix/ViewModels/RelationshipMatrixViewModel.cs [hunk 2]
```diff
@@ -787,6 +880,25 @@ private void ComputeEditInspectCanExecute()
             this.CanEditSourceX = vm.SourceX != null && this.PermissionService.CanWrite(vm.SourceX);
             this.CanInspectSourceY = vm.SourceY != null;
             this.CanInspectSourceX = vm.SourceX != null;
+
+
```
Comment (alexatrhea) [inline RelationshipMatrix/ViewModels/RelationshipMatrixViewModel.cs]: Same here.

Comment (alexatrhea) [inline RelationshipMatrix/ViewModels/RelationshipMatrixViewModel.cs]: Same here.

**Comment by lxatrhea** [comment]
Removed the white lines. I'll finish the request for now.

**Pull request merged**